### DISCUSSION TITLE
How do you control the "strength" of the float?

### DISCUSSION THREAD

**Discussion by musjj** [idanarye/bevy-tnua#55] [Q&A] [answered]
I'm using `avian2d` + `bevy-tnua` and I find myself needing to use a huge gravity value:

```rust
Gravity(Vec2::new(0., -98. * 12.0))
```

Because my game's world is really small (a low-res and scaled-up, pixel-art game). But the huge gravity value is causing the floating controller to constantly dip down and float up constantly, even when idle.

Is there a way to control the "strength" of the float to prevent this? Or is there a better approach for dealing with low resolution worlds in general?

I'm also having trouble controlling the initial velocity of `TnuaBuiltinWalk`. Even with huge `TnuaBuiltinWalk::desired_velocity` values, the initial walking speed remains low. I'm guessing this is related to the issue above.

**Comment by idanarye** [comment, marked as answer]
Try tweaking with [`spring_strengh`](https://docs.rs/bevy-tnua/0.19.0/bevy_tnua/builtins/struct.TnuaBuiltinWalk.html#structfield.spring_strengh) (and maybe [`spring_dampening`](https://docs.rs/bevy-tnua/0.19.0/bevy_tnua/builtins/struct.TnuaBuiltinWalk.html#structfield.spring_dampening))

But I suspect the real culprit is [`cling_distance`](https://docs.rs/bevy-tnua/0.19.0/bevy_tnua/builtins/struct.TnuaBuiltinWalk.html#structfield.cling_distance). The default works nice for regular-scaled worlds, but it's possible that it's too low for your scale.
